pkg/providers/pricing/pricing.go

```diff
@@ -80,6 +79,19 @@ type zonal struct {
 	prices       map[string]float64
 }
 
+func combineZonalPricing(pricingData ...zonal) zonal {
+	c := newZonalPricing(0)
```

Comment:
nit: where does `c` come from? Can we use `z` for `zonal`?
```suggestion
	z := newZonalPricing(0)
```